module-4/metaanalyse/metaanalyse.py

```diff
@@ -0,0 +1,81 @@
+import os
+import csv
+import argparse
+from PyPDF2 import PdfReader
+
+def extract_pdf_metadata(file_path):
+    metadata = {
+        'File Name': os.path.basename(file_path),
+        'Title': 'undefined',
+        'Author': 'undefined',
+        'Creator': 'undefined',
+        'Created': 'undefined',
+        'Modified': 'undefined',
+        'Subject': 'undefined',
+        'Keywords': 'undefined',
+        'Description': 'undefined',
+        'Producer': 'undefined',
+        'PDF Version': 'undefined'
+    }
+    try:
+        reader = PdfReader(file_path)
+        info = reader.metadata
+
+        metadata['Title'] = info.get('/Title', 'undefined') or 'undefined'
+        metadata['Author'] = info.get('/Author', 'undefined') or 'undefined'
+        metadata['Creator'] = info.get('/Creator', 'undefined') or 'undefined'
+        metadata['Created'] = info.get('/CreationDate', 'undefined') or 'undefined'
+        metadata['Modified'] = info.get('/ModDate', 'undefined') or 'undefined'
+        metadata['Subject'] = info.get('/Subject', 'undefined') or 'undefined'
+        metadata['Keywords'] = info.get('/Keywords', 'undefined') or 'undefined'
+        metadata['Description'] = info.get('/Description', 'undefined') or 'undefined'
+        metadata['Producer'] = info.get('/Producer', 'undefined') or 'undefined'
+        
+        metadata['PDF Version'] = reader.trailer.get('/Version', 'undefined') or 'undefined'
+    except Exception as e:
+        print(f'Error extracting metadata from {file_path}: {e}')
+    return metadata
+
+def process_single_file(file_path, csv_path):
+    metadata = extract_pdf_metadata(file_path)
+    with open(csv_path, 'w', newline='', encoding='utf-8') as csv_file:
+        fieldnames = ['File Name', 'Title', 'Author', 'Creator', 'Created', 'Modified', 'Subject', 'Keywords', 'Description', 'Producer', 'PDF Version']
+        writer = csv.DictWriter(csv_file, fieldnames=fieldnames, delimiter=';')
+        writer.writeheader()
+        writer.writerow(metadata)
+    print(f'Metadata saved to {csv_path}')
+
+def process_directory(directory_path, csv_path):
+    metadata_list = []
+    for root, dirs, files in os.walk(directory_path):
+        for file in files:
+            if file.lower().endswith('.pdf'):
+                file_path = os.path.join(root, file)
+                metadata = extract_pdf_metadata(file_path)
+                metadata_list.append(metadata)
+
+    with open(csv_path, 'w', newline='', encoding='utf-8') as csv_file:
+        fieldnames = ['File Name', 'Title', 'Author', 'Creator', 'Created', 'Modified', 'Subject', 'Keywords', 'Description', 'Producer', 'PDF Version']
+        writer = csv.DictWriter(csv_file, fieldnames=fieldnames, delimiter=';')
+        writer.writeheader()
+        for metadata in metadata_list:
+            writer.writerow(metadata)
```

Comment:
this is more or less the same code than above for single file processing -> extract the code into a reusable function that can create your result file for one or more files where metadata should be extracted